packages/server/src/fhir/precommit.ts

```diff
@@ -35,9 +39,13 @@ export function isPreCommitSubscription(subscription: WithId<Subscription>): boo
 export async function preCommitValidation<T extends Resource>(
   author: Reference,
   project: WithId<Project> | undefined,
-  resource: T,
+  resource: WithId<T>,
   interaction: BackgroundJobInteraction
 ): Promise<T | boolean | undefined> {
+  if (interaction === 'delete' && !project?.superAdmin) {
```

Comment:
The description of `project` is `The project to which the resource belongs. Must be passed separately because this is called before the resource meta is assigned.` whereas I think the intent here is to allow super admins to bypass this check, right? Should the signature of `preCommitValidation` be changed/amended? Maybe something like:

```typescript
export async function preCommitValidation<T extends Resource>(
  repo: Repository,
  resource: T,
  interaction: BackgroundJobInteraction
): Promise<T | boolean | undefined> {
```

It seems like that would tighten up the abstraction and would let this line be `if (interaction === 'delete' && !repo.isSuperAdmin()) {`

Reply:
Great suggestion, I think passing the whole `Repository` is much cleaner and didn't require much in the way of additional changes